validation/validation.go

```diff
@@ -20,32 +20,36 @@ import (
 // User name section:
 // ——————————————————
 // Grouped(
-//   first character set ensures that email starts with either a valid character (one or more)
-//   Grouped(second character set ensures characters are present after an escaped `.` (one or more)) (zero or more)
-//   OR
-//   Grouped(third character set matching permitted ascii hex characters OR a fourth character set matching a different set of permitted ascii hex characters) (zero or more)
-//)
+//
+//	first character set ensures that email starts with either a valid character (one or more)
+//	Grouped(second character set ensures characters are present after an escaped `.` (one or more)) (zero or more)
+//	OR
+//	Grouped(third character set matching permitted ascii hex characters OR a fourth character set matching a different set of permitted ascii hex characters) (zero or more)
+//
+// )
 //
 // Domain name section:
 // ————————————————————
 // Grouped(
-//   first character set ensures that email domain section starts with an alphanumeric character
-//   Grouped(second character set ensures alphanumeric/hyphen (`-`) characters followed by an alphanumeric character with a non-greedy match to first escaped `.` encountered) (one or more)
-//   Third character set ensures that character immediately after `.` is an alphanumeric character
-//   Grouped(fourth character set ensures alphanumeric/hyphen (`-`) characters) (zero or more) followed by fifth character set, ensuring domain name section ends in an alphanumeric character)
-//)
+//
+//	first character set ensures that email domain section starts with an alphanumeric character
+//	Grouped(second character set ensures alphanumeric/hyphen (`-`) characters followed by an alphanumeric character with a non-greedy match to first escaped `.` encountered) (one or more)
+//	Third character set ensures that character immediately after `.` is an alphanumeric character
+//	Grouped(fourth character set ensures alphanumeric/hyphen (`-`) characters) (zero or more) followed by fifth character set, ensuring domain name section ends in an alphanumeric character)
+//
+// )
 //
 // Example:
 // ————————
 // my.name@myself.com - match found - valid
 // .my.name@myself.com - match not found - invalid
 // my.name@myself.com. - match not found - invalid
-var emailRegex = regexp.MustCompile(`^(?:[a-z0-9!#$%&'*+/=?^_` + "`" + `{|}~-]+(?:\.[a-z0-9-!#$%&'*+/=?^_` + "`" + `{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?)$`)
+var emailRegex = regexp.MustCompile(`^(?:[a-z0-9!#$%&'*+/=?^_` + "`" + `{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_` + "`" + `{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?)$`)
```

Comment:
Why has this been changed? Do we no longer want to allow a `-` in the first part anymore?